src/win32/fs.cpp

```diff
@@ -1648,11 +1648,33 @@ bool WinFileSystemAccess::issyncsupported(string *localpath, bool *isnetwork)
     fsname.resize(MAX_PATH * sizeof(WCHAR));
 
     if (GetVolumePathNameW((LPCWSTR)localpath->data(), (LPWSTR)path.data(), MAX_PATH)
-        && GetVolumeInformationW((LPCWSTR)path.data(), NULL, 0, NULL, NULL, NULL, (LPWSTR)fsname.data(), MAX_PATH)
-        && !memcmp(fsname.data(), VBoxSharedFolderFS, sizeof(VBoxSharedFolderFS)))
+        && GetVolumeInformationW((LPCWSTR)path.data(), NULL, 0, NULL, NULL, NULL, (LPWSTR)fsname.data(), MAX_PATH))
     {
-        LOG_warn << "VBoxSharedFolderFS is not supported because it doesn't provide ReadDirectoryChanges() nor unique file identifiers";
-        result = false;
+        if (!memcmp(fsname.data(), VBoxSharedFolderFS, sizeof(VBoxSharedFolderFS)))
+        {
+            LOG_warn << "VBoxSharedFolderFS is not supported because it doesn't provide ReadDirectoryChanges() nor unique file identifiers";
+            if (syncError) *syncError = VBOXSHAREDFOLDER_UNSUPPORTED;
+            result = false;
+        }
+
+        // TODO: have these checks for !windows too
+        if ((!memcmp(fsname.data(), L"FAT", 6) || !memcmp(fsname.data(), L"exFAT", 10)))
```

Comment:
I understand that if a FS is VirtualBoxSharedFolderFS, it is not FAT nor HGFS, so here an `if else` sounds better, IMO.